### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by vikfil**

### CODE REVIEW

**Review by volodymyr-chornyi — APPROVED** [review]

File: src/main/java/com/softserve/academy/event/config/HibernateConfig.java [hunk 1]
```diff
@@ -28,7 +28,8 @@
 
     private static final String ENTITY_PACKAGE = "hibernate.entity.package";
 
-    private final Environment environment;
+    //@Autowired
+    private Environment environment;
```
Comment (clevercattv) [inline src/main/java/com/softserve/academy/event/config/HibernateConfig.java]: ?

File: src/main/java/com/softserve/academy/event/controller/LoginController.java [hunk 2]
```diff
@@ -1,21 +1,86 @@
...
+
+    @Autowired
+    private ApplicationEventPublisher eventPublisher;
+
+
+    @Autowired
+    EmailService emailService;
+
+    @Autowired
+    Environment env;
```
Comment (clevercattv) [inline src/main/java/com/softserve/academy/event/controller/LoginController.java]: Autowired set in constructor

File: src/main/java/com/softserve/academy/event/controller/LoginController.java [hunk 3]
```diff
@@ -1,21 +1,86 @@
...
+    public ResponseEntity registerUserAccount(@RequestBody UserDto accountDto, HttpServletRequest request) throws EmailExistException {
+        UserDto registered = userMapper.userToDto(userService.newUserAccount(userMapper.userDtoToUser(accountDto)));;
+        eventPublisher.publishEvent(new RegistrationCompleteEvent(userMapper.userDtoToUser(registered), request.getLocale(), getAppUrl(request)));
+        return new ResponseEntity(HttpStatus.CREATED);
+    }
+
+
+    @GetMapping(value = "/registrationConfirm")
+    public ResponseEntity confirmRegistration( @RequestParam("token")String token, HttpServletRequest request)  {
+        Locale locale = request.getLocale();
```
Comment (clevercattv) [inline src/main/java/com/softserve/academy/event/controller/LoginController.java]: Can't see request and locale using..

File: src/main/java/com/softserve/academy/event/entity/PasswordResetToken.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    public User getUser() {
+        return user;
+    }
+
+    public void setUser(final User user) {
+        this.user = user;
+    }
+
+    public Date getExpiryDate() {
+        return expiryDate;
```
Comment (clevercattv) [inline src/main/java/com/softserve/academy/event/entity/PasswordResetToken.java]: Lombok : @Getter and @Setter?

File: src/main/resources/database.properties [hunk 5]
```diff
@@ -1,19 +1,20 @@
 hibernate.connection.driver_class=org.postgresql.Driver
 hibernate.connection.url=jdbc:postgresql://localhost:5432/softserve_event_table
 hibernate.connection.username=postgres
-hibernate.connection.password=root
-hibernate.connection.provider_class=org.hibernate.connection.C3P0ConnectionProvider
-#hibernate.connection.pool_size=30
+hibernate.connection.password=admin
```
Comment (clevercattv) [inline src/main/resources/database.properties]: admin?

File: src/main/java/com/softserve/academy/event/entity/PasswordResetToken.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    public void setExpiryDate(final Date expiryDate) {
+        this.expiryDate = expiryDate;
+    }
+
+    private Date calculateExpiryDate(final int expiryTimeInMinutes) {
+        final Calendar cal = Calendar.getInstance();
+        cal.setTimeInMillis(new Date().getTime());
```
Comment (clevercattv) [inline src/main/java/com/softserve/academy/event/entity/PasswordResetToken.java:49]: new Date().getTime() == System.GetCurrentTimeMillis

File: src/main/java/com/softserve/academy/event/entity/PasswordResetToken.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        cal.add(Calendar.MINUTE, expiryTimeInMinutes);
+        return new Date(cal.getTime().getTime());
```
Comment (clevercattv) [inline src/main/java/com/softserve/academy/event/entity/PasswordResetToken.java]: cal.getTime().getTime() == cal.getTimeInMillis() ?

**Review by clevercattv — CHANGES_REQUESTED** [review]

File: src/main/resources/database.properties (hunk 5 above)
Comment (vikfil) [reply src/main/resources/database.properties]: root)

File: src/main/java/com/softserve/academy/event/controller/LoginController.java (hunk 2 above)
Comment (vikfil) [reply src/main/java/com/softserve/academy/event/controller/LoginController.java]: Ok

File: src/main/java/com/softserve/academy/event/config/HibernateConfig.java (hunk 1 above)
Comment (vikfil) [reply src/main/java/com/softserve/academy/event/config/HibernateConfig.java]: Ok

File: src/main/java/com/softserve/academy/event/entity/PasswordResetToken.java (hunk 4 above)
Comment (vikfil) [reply src/main/java/com/softserve/academy/event/entity/PasswordResetToken.java]: Ok

**Review by clevercattv — APPROVED** [review]

File: src/main/java/com/softserve/academy/event/entity/PasswordResetToken.java (hunk 7 above)
Comment (vikfil) [reply src/main/java/com/softserve/academy/event/entity/PasswordResetToken.java]: In this case   cal.getTimeInMillis() is more prefer method.

File: src/main/java/com/softserve/academy/event/entity/PasswordResetToken.java (hunk 6 above)
Comment (vikfil) [reply src/main/java/com/softserve/academy/event/entity/PasswordResetToken.java:49]: In this case, those methods are equivalent

File: src/main/java/com/softserve/academy/event/controller/LoginController.java (hunk 3 above)
Comment (vikfil) [reply src/main/java/com/softserve/academy/event/controller/LoginController.java]: Deleted

**Review by Qwe1999 — APPROVED** [review]

**Review by JuliaGrama21 — APPROVED** [review]

**Pull request merged**